Documentation/typo3cms/Developers/Index.rst

```diff
@@ -0,0 +1,196 @@
+-----------------
+
+.. _resources-developers-extension-general-references:
+
+Reference Guides in the official TYPO3 documentation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Look at the official list of references: :ref:`List of TYPO3 references<references>`
+
+For developing you will be interested in all the manuals listed there, especially the
+:ref:`TYPO3 Core API <t3coreapi:start>`.
+
+Additionally, see the references for TYPO3 Fluid ViewHelpers.
+
+* :ref:`Reference for Fluid ViewHelpers <t3extbasebook:viewhelpers>`
+
+
+.. _resources-developers-extension-general-videos:
+
+YouTube Videos
+~~~~~~~~~~~~~~
+
+* `TYPO3 YouTube Channel <https://www.youtube.com/channel/UCwpl8LY9Tr3PB26Kk2FYW_w>`__
+
+Unofficial Resources
+~~~~~~~~~~~~~~~~~~~~
+
+These are not "official" TYPO3 resources, but they have been reported to be very
+useful and sometimes provide more up-to-date information than the official docs.
+
+Blogs:
+
+.. note to editors: please only add blogs specifically dedicated to programming for TYPO3 or
+.. link to specific blog articles about programming for TYPO3
+
+* Daniel Goerz: `Usetypo3 <https://usetypo3.com/>`__  This might be the best resource for getting
+  started, for example look at `Good practices in Extensions <https://usetypo3.com/good-practices-in-extensions.html>`__,
+  and `24 Tips & Tricks for Fluid <https://usetypo3.com/24-fluid-tips.html>`__
+  Marcus Schwemer: `TYPO3 WORX <https://typo3worx.eu>`__
+* Andreas Fernandez (member of TYPO3 core team): `SCRIPTING-BASE <https://scripting-base.de/blog.html>`__
+  provides some in-depth information for example about Ajax and RequireJS and also very useful
+  instructions for :ref:`migrating-extensions`.
+* Helmut Hummel (member of TYPO3 core team): `doc_core_insight <https://usetypo3.com/>`__ Helmut's
+  resources are especially useful, if you are looking at information about composer and TYPO3,
+  executing commands from the console (typo3_console), caching, autoloading etc. It is often in-depth
+  information that helps to understand things, not always suitable for beginners.
+
+
+Specific Resources
+------------------
+
+.. _migrating-extensions:
+
+Migrating extensions
+~~~~~~~~~~~~~~~~~~~~
+
+You may be maintaining a TYPO3 installation and need to update to the
+next major TYPO3 version. TYPO3 already takes care of everything you
+need to do very nicely in the Upgrade Wizard. Third party extensions
+can also be updated easily.
+
+If you have developed some extensions yourself, you will want to make
+sure, that they will be ready for the next major TYPO3 version *before*
+you update!
+
+For this, the deprecation log and :ref:`Extension Scanner <extension-scanner>`
+(Extension Scanner since TYPO3 version 9) is very valuable.
+
+Additionally, look at these resources:
+
+* Check the `Official Changelogs <https://docs.typo3.org/typo3cms/extensions/core/latest/>`__
+  for "Breaking changes" for your target version. Also checkout the "Changes for Developers"
```

Comment:
In this context, it should be `check out` instead of `checkout`, as far as I understand.